tests/parametric/test_otel_span_methods.py

```diff
@@ -330,21 +588,797 @@ def test_otel_set_span_status_ok(self, test_agent, test_library):
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_message_producer(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `messaging.system + "." + messaging.operation` when:
+            - Span kind is set to Producer
+            - messaging.system is set to something
+            - messaging.operation is set to something
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_PRODUCER) as span:
+                span.set_attributes({"messaging.system": "kafka"})
+                span.set_attributes({"messaging.operation": "publish"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "kafka.publish"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_message_server(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `messaging.system + "." + messaging.operation` when:
+            - Span kind is set to Server
+            - messaging.system is set to something
+            - messaging.operation is set to something
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_SERVER) as span:
+                span.set_attributes({"messaging.system": "kafka"})
+                span.set_attributes({"messaging.operation": "receive"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "kafka.receive"
+        assert root_span["resource"] == "otel_span_name"
```

Comment:
You could consider creating a shared test the just accepts a span_kind that's different, to avoid code duplication. There's probably a few places in these tests where doing that would make it a bit easier to read.

I actually wonder if you could make a shared test utility for all/most of these tests. Basically the utility will create, end, and verify a span given the `span_kind` (optional), a map of attributes (optional), and the expected span name to verify. This might make a lot of these tests easy to understand at a glance.

Reply:
That's a great suggestion, I'll see if I can make a utility for this

Reply:
I've parametrized and combined the message ones with just a `@parametrize` for each of the span kinds in 6560381683418f94feec1ec36f7632a4450b4078

I can continue looking into reducing the rest of the duplication, but figured that seemed like an easy change to make

Reply:
I've made a helper method for the operation name tests in 6f1b74be2c55e871ee16c45a35a16906b00e23f3 and combined the other span kind ones with a `@parametrize`

Reply:
I still think most of the naming tests can be summarise in a single parametrised test.
That is what I did for Java tracer tests and it feels easier to ready: https://github.com/DataDog/dd-trace-java/pull/6104/files#diff-031f3e359dd7470cc178e857ab513ec88199f0365848a0478317cc2e2f252439R54-R90

Reply:
@PerfectSlayer Yeah that seems to help a lot and will probably make it easier to add more cases if needed -> fba65e977761ab88aebe20214fbb6aa63d10060f